### DISCUSSION TITLE
Use custom select option labels for `ModelMultipleChoiceFilter` instead of `__str__(self)`?

### DISCUSSION THREAD

**Discussion by tombreit** [carltongibson/django-filter#1656] [Q&A] [answered]
I am using the `ModelMultipleChoiceFilter`, but like to customize the  text of the options (here: instances of `models.Author`) displayed to the user? Currently these options/labels  are the models `__str__(self)` method.

As I already provide a custom queryset, I would like to provide a custom "label template" or overwrite the default select option labels:

```python
def authors_site_scoped(request):
    return Author.objects.filter(...).distinct()

class PublicationFilter(django_filters.FilterSet):

    authors_filter = django_filters.ModelMultipleChoiceFilter(
        field_name="authors",
        queryset=authors_site_scoped, 
    )

    class Meta:
        model = Publication
        form = PublicationFilterForm
        fields = (
            "authors_filter",
        )
```

Is that possible? And if: how :-)?

**Comment by carltongibson** [comment, marked as answer]
https://docs.djangoproject.com/en/5.0/ref/forms/fields/#django.forms.ModelChoiceField.iterator

**Reply by tombreit** [reply]
Thank you very much - works perfectly! And for my future self: 

```python
class AuthorModelMultipleChoiceField(forms.ModelMultipleChoiceField):
    def label_from_instance(self, obj):
        return obj.get_select_label  # or return f"{obj.last_name}, {obj.first_name}"

class AuthorModelMultipleChoiceFilter(django_filters.ModelMultipleChoiceFilter):
    field_class = AuthorModelMultipleChoiceField

class PublicationFilter(django_filters.FilterSet):
    authors_filter = AuthorModelMultipleChoiceFilter(
        field_name="authors",
    )
```


Unfortunately, it was not possible to set the `field_class=AuthorModelMultipleChoiceField` directly for `authors_filter` (`Field.__init__() got an unexpected keyword argument 'field_class'`).
